### PR TITLE
Feedback tts

### PR DESCRIPTION
**Pull request by anthonyray**

### CODE REVIEW

File: snipssonos/shared/response_object.py [hunk 1]
```diff
@@ -12,9 +12,10 @@ class ResponseFailure(object):
     PARAMETERS_ERROR = 'ParametersError'
     SYSTEM_ERROR = 'SystemError'
 
-    def __init__(self, type_, message):
+    def __init__(self, type_, message, exception=None):
         self.type = type_
         self.message = self._format_message(message)
+        self.exception = exception
```
Comment (paufabregat) [inline snipssonos/shared/response_object.py:18]: Great you added this. I did not realize I was missing it until today.

File: snipssonos/adapters/tts_sentence_adapter.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from snipssonos.shared import feedback
+from snipssonos.exceptions import ExternalDeviceDiscoveryUnreachable
```
Comment (paufabregat) [inline snipssonos/adapters/tts_sentence_adapter.py]: This is pointing to nowhere.

File: snipssonos/adapters/tts_sentence_adapter.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    def language(self):
+        return self._language
+
+    @language.setter
+    def language(self, lang):
+        if not (lang in self.SUPPORTED_LANGUAGES):
+            class_name = self.__class__.__name__
+            raise AttributeError(
+                "Tried to assign an unsupported language to the language property of {}".format(class_name))
+        else:
```
Comment (paufabregat) [inline snipssonos/adapters/tts_sentence_adapter.py:21]: Please, get rid of the else you do not need it. If the previous condition passes then this part of the code is never reached anyways.

File: snipssonos/adapters/tts_sentence_adapter.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            self._language = lang
+
+    def from_response_object(self, response_object):
+        if response_object: # ResponseSuccess
+            return response_object.message
+        else: # ResponseFailure
+            if isinstance(response_object.exception, ExternalDeviceDiscoveryUnreachable):
+                return feedback.FR_TTS_DEVICE_DISCOVERY_SERVICE_UNREACHABLE
+            else:
+                return "" # TODO : this is not correct.
```
Comment (paufabregat) [inline snipssonos/adapters/tts_sentence_adapter.py]: Would be correct? Just a generic error message no?

File: action-sonos.py [hunk 5]
```diff
@@ -2,34 +2,38 @@
...
 from snipssonos.helpers.snips_config_parser import read_configuration_file
 from snipssonos.use_cases.volume.up import VolumeUpUseCase
 from snipssonos.use_cases.volume.down import VolumeDownUseCase
 from snipssonos.use_cases.volume.set import VolumeSetUseCase
 from snipssonos.use_cases.mute import MuteUseCase
-from snipssonos.use_cases.play.track import PlayTrackUseCase
-from snipssonos.use_cases.play.artist import PlayArtistUseCase
-from snipssonos.use_cases.play.music import PlayMusicUseCase
-from snipssonos.use_cases.play.playlist import PlayPlaylistUseCase
-from snipssonos.use_cases.play.album import PlayAlbumUseCase
```
Comment (paufabregat) [inline action-sonos.py]: this should be back, right now it is pointing to the wrong places

File: action-sonos.py [hunk 6]
```diff
@@ -2,34 +2,38 @@
...
+from snipssonos.use_cases.next_track import NextTrackUseCase
 
 from snipssonos.adapters.request_adapter import VolumeUpRequestAdapter, PlayTrackRequestAdapter, \
     PlayArtistRequestAdapter, VolumeSetRequestAdapter, VolumeDownRequestAdapter, ResumeMusicRequestAdapter, \
     SpeakerInterruptRequestAdapter, MuteRequestAdapter, PlayPlaylistRequestAdapter, PlayAlbumRequestAdapter, \
     PlayMusicRequestAdapter, NextTrackRequestAdapter
-from snipssonos.services.node.device_discovery_service import NodeDeviceDiscoveryService
-from snipssonos.services.node.device_transport_control import NodeDeviceTransportControlService
-from snipssonos.services.node.music_playback_service import NodeMusicPlaybackService
-from snipssonos.services.spotify.music_search_service import SpotifyMusicSearchService
```
Comment (paufabregat) [inline action-sonos.py]: Same here

File: action-sonos.py [hunk 7]
```diff
@@ -2,34 +2,38 @@
 # -*-: coding utf-8 -*-
 
 import logging
+import io
 
 from hermes_python.hermes import Hermes
+from hermes_python.ontology import *
```
Comment (paufabregat) [inline action-sonos.py]: Not used and same for `import io`

File: action-sonos.py [hunk 8]
```diff
@@ -38,11 +42,13 @@
 MOPIDY_HOST = HOSTNAME
 
 # Logging config
-logging.basicConfig(level=logging.INFO)
+logging.basicConfig(level=logging.DEBUG, format='%(asctime)s - %(levelname)s - %(message)s')
+
```
Comment (paufabregat) [inline action-sonos.py]: Check PEP8

File: action-sonos.py [hunk 9]
```diff
@@ -233,4 +254,5 @@ def playMusic_callback(hermes, intentMessage):
             .subscribe_intent("muteSound4", mute_callback) \
             .subscribe_intent("resumeMusic4", resumeMusic_callback) \
             .subscribe_intent("speakerInterrupt4", speakerInterrupt_callback) \
+            .subscribe_intent("nextSong4", nextSong_callback) \
```
Comment (paufabregat) [inline action-sonos.py:237]: We should define the intent names in a file and import from it, I see they are changing constantly with the new versions of the assistants. So at least we have a place where we centralize this. 
I will open an issue if you agree with this.

**Review by paufabregat — CHANGES_REQUESTED** [review]

File: snipssonos/adapters/tts_sentence_adapter.py (hunk 2 above)
Comment (anthonyray) [reply snipssonos/adapters/tts_sentence_adapter.py]: Good catch !

File: snipssonos/adapters/tts_sentence_adapter.py (hunk 3 above)
Comment (anthonyray) [reply snipssonos/adapters/tts_sentence_adapter.py:21]: Actually, we need this. If the `lang` parameter is in the list of the `SUPPORTED_LANGUAGE` array, we set it as the `_language` attribute of the class.

File: snipssonos/adapters/tts_sentence_adapter.py (hunk 4 above)
Comment (anthonyray) [reply snipssonos/adapters/tts_sentence_adapter.py]: I now realize how rough was my P.R ^^! 
My goal here is to list the possible exceptions and react accordingly.

File: action-sonos.py (hunk 5 above)
Comment (anthonyray) [reply action-sonos.py]: yup ! sorry !

File: action-sonos.py (hunk 7 above)
Comment (anthonyray) [reply action-sonos.py]: Will fix

File: action-sonos.py (hunk 9 above)
Comment (anthonyray) [reply action-sonos.py:237]: Yes, this is a good idea. Open up an issue. 
We should do the same thing for TTS feedback strings btw.

File: snipssonos/adapters/tts_sentence_adapter.py (hunk 3 above)
Comment (paufabregat) [reply snipssonos/adapters/tts_sentence_adapter.py:21]: I meant this: 
~~~python
if not (lang in self.SUPPORTED_LANGUAGES):
       class_name = self.__class__.__name__
       raise AttributeError(
       "Tried to assign an unsupported language to the language property of {}".format(class_name))

self._language = lang
~~~

**Pull request merged**